cli/src/error.rs

```diff
@@ -20,6 +25,31 @@ pub enum Error {
     Format {
         error: crate::format::FormatError,
     },
+    /// An invalid invocation of the CLI that couldn't be caught by the simple parsing provided by
+    /// clap.
+    CliUsage {
+        program: Program<CBNCache>,
+        error: CliUsageError,
+    },
+}
+
+impl<FileId> IntoDiagnostics<FileId> for CliUsageError {
+    fn into_diagnostics(
+        self,
+        _files: &mut Files<String>,
+        _stdlib_ids: Option<&Vec<FileId>>,
+    ) -> Vec<Diagnostic<FileId>> {
+        match self {
+            CliUsageError::InvalidOverride { path } => {
+                vec![Diagnostic::error()
+                    .with_message(format!("invalid override: unknown field `{path}`"))
+                    .with_notes(vec![format!(
+                        "`{path}` doesn't refer to record field accessible from the root of the \
+                        configuration and thus can't be the target of `--override`."
```

Comment:
Right, and even more because `clap` does this automatically for arguments, hence fields with a definition should benefit from it already, but not overridable fields, which isn't very consistent.